data_structures/heap/min_heap.py

```diff
@@ -0,0 +1,154 @@
+# Min head data structure
+# with decrease key functionality - in O(log(n)) time
+
+
+class Node:
+    def __init__(self, val, name):
+        self.val = val
+        self.name = name
+
+    def __str__(self):
+        return self.name
```

Comment:
__return f"{self.\_\_class\_\_.\_\_name\_\_}({self.name}, {self.value})"__
The Python convention is that \_\_str\_\_() should print out exactly what it would take to build a new instance.